tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc

```diff
@@ -254,80 +254,92 @@ int TRTEngineOp::GetEngineBatch(tensorflow::OpKernelContext* ctx) {
       cached_engine_batches_.push_back(num_batch);
       VLOG(1) << "Running with batch size " << num_batch;
     } else {
-      string s("Engine buffer is full. buffer limit= ");
-      StrAppend(&s, max_cached_engines_, ", current entries= ");
-      for (auto i : cached_engine_batches_) StrAppend(&s, i, ", ");
-      StrAppend(&s, "Requested batch= ", num_batch);
-      LOG(ERROR) << s;
-      ctx->SetStatus(tensorflow::errors::ResourceExhausted(
-          "Requested batch size is not available and engine cache is full"));
+      string msg =
+          StrCat("Engine buffer is full. buffer limit=", max_cached_engines_,
+                 ", current entries=");
+      for (auto i : cached_engine_batches_) StrAppend(&msg, i, ",");
+      StrAppend(&msg, "Requested batch=", num_batch);
+      LOG(WARNING) << msg;
       return -1;
     }
   }
   return smallest_engine;
 }
 
-void TRTEngineOp::ComputeAsync(tensorflow::OpKernelContext* ctx,
-                               tensorflow::AsyncOpKernel::DoneCallback done) {
+void TRTEngineOp::ComputeAsync(OpKernelContext* ctx,
+                               AsyncOpKernel::DoneCallback done) {
   auto helper = new AsyncHelper(done);
   tensorflow::core::ScopedUnref sc(helper);
   if (calibration_mode_) {
     ExecuteCalibration(ctx, helper);
     return;
   }
   const int smallest_engine = GetEngineBatch(ctx);
-  if (smallest_engine < 0) return;  // GetEngineBatch already set the status.
+  if (smallest_engine < 0) {
```

Comment:
Question here is whether to abort or continue. I would say abort is a better option since more engines than user expected is required.

Reply:
This one is the reason why I issue this PR. In practice we encounter a problem where the first dim is not batch size but the segment gets converted, and another segment with correct first dim as batch size gets converted, too. In that case, the batch size dim doesn't match and one of the engine will fail since we allow single cache entry. In that case we should retry.

I'm not saying that we should not fix the batch dim problem, on the contrary I strongly suggest to fix it (although I don't know how). But since there is always a possibility to have bugs in the conversion regardless how many tests we added, I think we should gracefully retry on non-critical errors.